archunit/src/main/java/com/tngtech/archunit/library/modules/ModuleDependency.java

```diff
@@ -0,0 +1,126 @@
+     */
+    @PublicAPI(usage = ACCESS)
+    public ArchModule<DESCRIPTOR> getOrigin() {
+        return origin;
+    }
+
+    /**
+     * @return The {@link ArchModule module} that this dependency targets
+     */
+    @PublicAPI(usage = ACCESS)
+    public ArchModule<DESCRIPTOR> getTarget() {
+        return target;
+    }
+
+    /**
+     * @return All the single {@link Dependency class dependencies} that form this {@link ModuleDependency},
+     *         i.e. all {@link Dependency dependencies} where the {@link Dependency#getOriginClass() origin class}
+     *         resides in the {@link #getOrigin() origin module} and the {@link Dependency#getTargetClass() target class}
+     *         resides in the {@link #getTarget() target module}.
+     */
+    @PublicAPI(usage = ACCESS)
+    public Set<Dependency> toClassDependencies() {
+        return classDependencies;
+    }
+
+    /**
+     * @return A textual representation of this {@link ModuleDependency} that can be used for textual reports.
+     */
+    @Override
+    @PublicAPI(usage = ACCESS)
+    public String getDescription() {
+        String classDependencyDescriptions = classDependencies.stream()
+                .map(HasDescription::getDescription)
+                .collect(joining(lineSeparator()));
+        return String.format("Module Dependency [%s -> %s]:%n%s", origin.getName(), target.getName(), classDependencyDescriptions);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(origin.getIdentifier(), target.getIdentifier());
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+        if (obj == null || getClass() != obj.getClass()) {
+            return false;
+        }
+        ModuleDependency<?> other = (ModuleDependency<?>) obj;
+        return Objects.equals(this.origin.getIdentifier(), other.origin.getIdentifier())
+                && Objects.equals(this.target.getIdentifier(), other.target.getIdentifier());
+    }
+
+    @Override
+    public String toString() {
+        return getClass().getSimpleName() + "{" +
+                "origin=" + origin +
+                ", target=" + target +
+                '}';
+    }
+
+    static <D extends ArchModule.Descriptor> Optional<ModuleDependency<D>> tryCreate(ArchModule<D> origin, ArchModule<D> target) {
+        Set<Dependency> classDependencies = filterDependenciesBetween(origin, target);
+        return !classDependencies.isEmpty()
+                ? Optional.of(new ModuleDependency<>(origin, target, classDependencies))
+                : Optional.empty();
+    }
+
+    private static Set<Dependency> filterDependenciesBetween(ArchModule<?> origin, ArchModule<?> target) {
+        return origin.getClassDependenciesFromSelf().stream()
+	                        .filter(dependency -> target.contains(dependency.getTargetClass()))
+	                        .collect(toImmutableSet());
```

Comment:
Just FYI: these lines somehow got an inconsistent indentation: 😉
```suggestion
                .filter(dependency -> target.contains(dependency.getTargetClass()))
                .collect(toImmutableSet());
```